textalyze.js

```diff
@@ -6,7 +6,7 @@ const fs = require('fs');
  * @returns {Map} counts - A Map containing the counts of the items in the input array
  */
 function itemCounts(array) {
-  return array.reduce((map, a) => {
+  return array.sort().reduce((map, a) => {
```

Comment:
Make sure to test that this is returning things alphabetically.

Alternatively, I wouldn't rely on `itemCounts` to return a `Map` with alphabetically sorted keys.  The reason is that something could modify the returned `Map` and add keys in a non-alphabetical order.

I'd either:

1. Implement a type that acts like map but guarantees the keys are in order (like `set` vs `sortedSet` in some languages).
2. Not assume anything about the order of the data in the map and instead iterate over the keys alphabetically when it comes time to display them.

For (2), that would mean `printHistogram` looks more like:

```javascript
Array.from(frequencies.keys()).sort().foreach(key => {
  const val = frequencies.get(key);
  ...
});
```

Reply:
It makes a lot more sense, thanks!